specs/enhance_gluster_cluster_sync_logic.adoc

```diff
@@ -0,0 +1,375 @@
+= Enhance gluster cluster synchronization logic as per recommendations
+
+The intent of this change is to accommodate additional details available as part
+of `get-state` output while syncing the gluster cluster details and store them
+in central store.
+
+Also instead of using `gstatus` for cluster health and volume utilization
+details, we need to user recommendations provided by gluster team.
+
+
+== Problem description
+
+This specification talks about various changes needed to be done under gluster
+cluster synchronization code to accomodate the additional details related to
+volume snapshots, geo-rep sessions.
+
+Also as `gtstaus` is quite a unstable and also few scenarios are not taken care
+of, using it for deciding the overall cluster status and volume utilization is
+not appropriate. Rather we need to write the aggregations logic to decide the
+cluster state based on peers status and volume status of the gluster trusted
+storage pool. Also the overall volume utilization should be calculated based on
+aggregations done for the participating bricks of the volumes.
+
+
+== Use Cases
+
+This addresses the synchronization of gluster cluster details like
+
+* Volume snapshot details
+* Add volume re-balance estimation time details
+* Volume geo-rep details
+* Client connections per volume
+* Identification of arbiter volumes in cluster (if any)
+* Instead of `gstatus` use aggregations across bricks utilization to calculate
+volume utilization
+* Instead of `gstatus` use peer status and volume status to aggregate and
+calculate the cluster status
+
+
+== Proposed change
+
+* Use `get-state detail` and `get-state volumeoptions` commands instead of plain
+`get-state` as first provides the detailed (e.g. client connections etc.)
+information for the volume and second is option to list all the volume options
+with default values
+
+* Use `get-state` output to figure out the volume snapshot details and update in
+central store
+
+* Calculate the estimation time for re-balance job. This should be largest
+value among the `time_left` values reported by get-state from different nodes of
+the cluster
+
+* Use `get-state` output for geo-rep session details like no. of geo-rep
+sessions and active geo-rep sessions and update per volume in central store
+
+* Figure out the client connections per brick from `get-state` output and
+aggregate for volume and update in central store
```

Comment:
Please document the aggregation strategy.

Also, I think there are certain pitfalls in using the client connections provided by gluster, such as no differentiation between actual clients and other processes. This needs to be documented clearly in the spec as it would be necessary for the documentation of this feature.